plexapi/library.py

```diff
@@ -436,21 +441,80 @@ def reload(self, key=None):
         return self._server.library.section(self.title)
 
     def edit(self, agent=None, **kwargs):
-        """ Edit a library (Note: agent is required). See :class:`~plexapi.library.Library` for example usage.
+        """ Edit a library. See :class:`~plexapi.library.Library` for example usage.
 
             Parameters:
+                agent (str, optional): The library agent.
                 kwargs (dict): Dict of settings to edit.
+
+            Example:
+
+                .. code-block:: python
+                LibrarySection.edit(**{'location': ['/path/1', 'path/2', '/path/3']})
         """
         if not agent:
             agent = self.agent
-        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(kwargs))
+
+        if kwargs['location']:
+            paths = kwargs.get('location')
+            for path in paths:
+                if not utils.pathExist(path):
+                    raise BadRequest('Path: %s does not exist.' % path)
+        params = list(kwargs.items())
+
+        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(params, doseq=True))
         self._server.query(part, method=self._server._session.put)
 
         # Reload this way since the self.key dont have a full path, but is simply a id.
         for s in self._server.library.sections():
             if s.key == self.key:
                 return s
 
+    def addLocations(self, location):
+        """ Add a location to a library.
+
+            Parameters:
+                location (str, list, or class): A single folder path, list of paths,
+                    or :class:`~plexapi.library.Location`.
+
+            Example:
+
+                .. code-block:: python
+                LibrarySection.addLocations('/path/1')
+                LibrarySection.addLocations(['/path/1', 'path/2', '/path/3'])
+                MovieLibrarySection.addLocations(ShowLibrarySection._locations()[0])
+        """
+        locations = self.locations
+        if isinstance(location, Location):
+            location = location.path
+        if isinstance(location, list):
+            locations += location
+        else:
+            locations.append(location)
+        self.edit(location=locations)
+
+    def removeLocations(self, location):
+        """ Remove a location from a library.
+
+            Parameters:
+                location (str, list, or class): A single folder path, list of paths,
+                     or :class:`~plexapi.library.Location`.
+            Example:
+
+                .. code-block:: python
+                LibrarySection.removeLocations('/path/1')
+                LibrarySection.removeLocations(['/path/1', 'path/2', '/path/3'])
+                LibrarySection.removeLocations(LibrarySection._locations()[0])
+        """
+        locations = self.locations
+        if isinstance(location, Location):
+            location = location.path
+        if isinstance(location, list):
+            locations -= location
```

Comment:
Lists don't have a `-` method. You need to remove them using a for loop.

Also need to catch the `ValueError` and raise `BadRequest` when trying to remove a path that doesn't exist.